nifi-mock/src/main/java/org/apache/nifi/util/MockProcessSession.java

```diff
@@ -496,7 +496,10 @@ public MockFlowFile putAllAttributes(FlowFile flowFile, final Map<String, String
         final MockFlowFile newFlowFile = new MockFlowFile(mock.getId(), flowFile);
         currentVersions.put(newFlowFile.getId(), newFlowFile);
 
-        newFlowFile.putAttributes(attrs);
+        Map<String, String> attrCopy = new HashMap<>();
+        attrCopy.putAll(attrs);
```

Comment:
Only change I'd make here is putting that into the constructor. Other than that LGTM.